### PR TITLE
Add building src

### PR DESCRIPTION
**Pull request by DamianDuy**
Added building the libuasync library from the provided sources and enabled using ```make cstyle``` for the sources.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	${CMAKE_CURRENT_SOURCE_DIR}/*.[ch]
+	${CMAKE_CURRENT_SOURCE_DIR}/include/*.h)
+
+set(srcs
+    future.c
+    mover.c
+    runtime.c
+)
+
+add_library(uasync SHARED ${srcs})
```
Comment (wlemkows) [inline src/CMakeLists.txt:20]: libuasync

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+add_cstyle(src
+	${CMAKE_CURRENT_SOURCE_DIR}/*.[ch]
+	${CMAKE_CURRENT_SOURCE_DIR}/include/*.h)
+
+add_check_whitespace(src
+	${CMAKE_CURRENT_SOURCE_DIR}/*.[ch]
+	${CMAKE_CURRENT_SOURCE_DIR}/include/*.h)
+
+set(srcs
```
Comment (wlemkows) [inline src/CMakeLists.txt:14]: srcs - > SOURCES

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    future.c
+    mover.c
+    runtime.c
+)
+
+add_library(uasync SHARED ${srcs})
+
+target_include_directories(uasync PRIVATE . include)
```
Comment (wlemkows) [inline src/CMakeLists.txt:22]: libuasync

File: src/CMakeLists.txt (hunk 2 above)
Comment (DamianDuy) [reply src/CMakeLists.txt]: Done.

File: src/CMakeLists.txt (hunk 3 above)
Comment (DamianDuy) [reply src/CMakeLists.txt:22]: From what I've read: ```"The actual file name of the library built is constructed based on conventions of the native platform (such as lib<name>.a or <name>.lib)."```

**Comment by wlemkows** [comment]
Moved to https://github.com/wlemkows/libuasync/pull/2

**Pull request closed without merging**